subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/api/types.go

```diff
@@ -158,3 +158,32 @@ type LazyPreemptionStatus struct {
 	// It was lazy preempted at PreemptionTime.
 	PreemptionTime meta.Time `json:"preemptionTime"`
 }
+
+type VirtualClusterList struct {
+	Items []VirtualCluster `json:"items"`
+}
+
+type VirtualCluster struct {
+	ObjectMeta `json:"metadata"`
+	Status     VirtualClusterStatus `json:"status"`
+}
+
+type VirtualClusterStatus struct {
+	VirtualClusterCapacity *ClusterCapacity `json:"virtualClusterCapacity"`
+}
+
+type ClusterCapacity struct {
+	Total                      int32 `json:"total"`
+	GuaranteedUsage            int32 `json:"guaranteedUsage"`
+	OpportunisticUsage         int32 `json:"opportunisticUsage"`
+	RemainingCapacityBestCase  int32 `json:"remainingCapacityBestCase"`
+	RemainingCapacityWorstCase int32 `json:"remainingCapacityWorstCase"`
+}
```

Comment:
Add detail comments for each field, as they are APIs

Reply:
Mentioned that the unit is GPU

---
In reply to: [368356556](https://github.com/microsoft/pai/pull/4153#discussion_r368356556) [](ancestors = 368356556)